jkube-kit/enricher/api/src/test/java/org/eclipse/jkube/kit/enricher/api/util/KubernetesResourceUtilTest.java

```diff
@@ -39,191 +40,97 @@
+  @SuppressWarnings("rawtypes")
+  @ParameterizedTest(name = "With {0}")
+  @MethodSource("validResource")
+  void getResource_fromFileAndValues(String testDesc, String file, Class clazz, String apiVersion, String kind, String metadata) throws IOException {
     // When
     final HasMetadata result = KubernetesResourceUtil.getResource(kubernetes, DEFAULT_RESOURCE_VERSIONING,
-        new File(fragmentsDir, "simple-rc.yaml"), "app");
+            new File(fragmentsDir, file), "app");
     // Then
     assertThat(result)
-        .isInstanceOf(ReplicationController.class)
-        .hasFieldOrPropertyWithValue("apiVersion", "v1")
-        .hasFieldOrPropertyWithValue("kind", "ReplicationController")
-        .hasFieldOrPropertyWithValue("metadata.name", "simple");
+            .isInstanceOf(clazz)
+            .hasFieldOrPropertyWithValue("apiVersion", apiVersion)
+            .hasFieldOrPropertyWithValue("kind", kind)
+            .hasFieldOrPropertyWithValue("metadata.name", metadata);
   }
 
-  @Test
-  public void getResource_withJsonFileAndVersionKindNameFromFile_shouldReturnValidResource() throws IOException {
-    // When
-    final HasMetadata result = KubernetesResourceUtil.getResource(kubernetes, DEFAULT_RESOURCE_VERSIONING,
-        new File(fragmentsDir, "simple-rc.json"), "app");
-    // Then
-    assertThat(result)
-        .isInstanceOf(ReplicationController.class)
-        .hasFieldOrPropertyWithValue("apiVersion", "v1")
-        .hasFieldOrPropertyWithValue("kind", "ReplicationController")
-        .hasFieldOrPropertyWithValue("metadata.name", "simple");
+  public static Stream<Arguments> validResource() {
+    return Stream.of(
+            Arguments.of("Yaml file and version and kind and name from file should return valid resource", "simple-rc.yaml", ReplicationController.class, "v1", "ReplicationController", "simple"),
+            Arguments.of("Json file and version kind name from file should return valid resource", "simple-rc.json", ReplicationController.class, "v1", "ReplicationController", "simple"),
+            Arguments.of("Value in name and file should be named from value", "named-svc.yaml", Service.class, "v1", "Service", "pong"),
+            Arguments.of("Value in name should be named named from value", "rc.yml", ReplicationController.class, "v1", "ReplicationController", "flipper"),
+            Arguments.of("No name in value and no name in file name should be named from app name", "svc.yml", Service.class, "v1", "Service", "app"),
+            Arguments.of("Value in kind and not in filename should get kind from value", "contains_kind.yml", ReplicationController.class, "v1", "ReplicationController", "flipper"),
+            Arguments.of("Kind from filename should get kind from filename", "job.yml", Job.class, "batch/v1", "Job", "app"),
+            Arguments.of("Network v1 ingress should load network v1 ingress", "network-v1-ingress.yml", Ingress.class, "networking.k8s.io/v1", "Ingress", "my-ingress")
```

Comment:
Like in other PRs, you're grouping too many tests.

Keep always in mind that tests should be readable and maintainable. For these tests, we end up not knowing what we're actually testing until you read the parameterized description. As stated in other reviews, you need to find the common denominator for the test cases to be able to parameterize the test with a description that uses place holders `fooBarMethodOrProcedureOrBehavior, with $baz, should do something expecting $qux.`